src/infrastructure/repositories/games.repository.js

```diff
@@ -134,4 +134,176 @@ export class GamesRepository {
       ])
       .toArray();
   }
+
+  async getTrendingGames(timePeriodInMs, returnAmount, minimumPlayers) {
+    await this.#dbClient.get("games").aggregate([
+      this.#getGamesWithMinimumXPlayers(minimumPlayers),
+      this.#addAveragePlayersTodayProperty(),
+      this.#addAveragePlayersAtCustomDayProperty(timePeriodInMs),
+      this.#getGamesWithAveragePlayersValues(),
+      this.#addPercentagePlayerIncreaseProperty(),
+      { $unset: "averagePlayersToday" },
+      { $unset: "averagePlayersAtCustomDay" },
+      {
+        $match: {
+          percentagePlayerIncrease: { $gt: 0 },
+        },
+      },
+      { $sort: { percentagePlayerIncrease: -1 } },
+      { $limit: returnAmount },
+    ]);
+  }
+
+  #getGamesWithMinimumXPlayers = (minimumPlayers) => {
+    return {
+      $match: {
+        $expr: {
+          $gt: [{ $arrayElemAt: ["$playerHistory.averagePlayers", -1] }, minimumPlayers],
+        },
+      },
+    };
+  };
+
+  #addAveragePlayersTodayProperty = () => {
```

Comment:
ok I get everything except those two big functions :), addAveragePlayersTodayProperty, addAveragePlayersAtCustomDayProperty...to many syntax which I don't know, but I can ask gpt to explain actually